File: kickstart/2022/coding_practice_with_kick_start_session_2/sample_problem/problem_statement/samples/sample.vb
Module kostka
  Sub test()
    ' Read integers N and M from the standard input as an array of strings.
    Dim NM() As String = Console.ReadLine().Split(" "c)
    ' Read N integers from the standard input and save them in the array of strings `C`.
    Dim C() As String = Console.ReadLine().Split(" "c)
    ' Declare a variable for sum and set it to 0.
    Dim sum As Integer = 0
    ' Loop through the array `C` and sum its values converted to integers.
    For Each i As String In C
      sum += Convert.ToInt32(i)
    Next
    ' Compute the value of sum modulo M.
    Dim modulo As Integer = sum Mod Convert.ToInt32(NM(1))
    ' Print the result onto the standard output.
    Console.WriteLine(modulo)
  End Sub

  Sub Main()
    ' Read the number of test cases.
    Dim T As Integer = Convert.ToInt32(Console.ReadLine())
    ' Loop over the number of test cases, starting from 1.
    For i As Integer = 1 To T
      ' Print case number
      Console.Write(String.Format("Case #{0}: ", i))
      ' and solve each test.
      test()
    Next
  End Sub
End Module

File: kickstart/2022/coding_practice_with_kick_start_session_2/sample_problem/problem_statement/samples/sample.kt
fun main(args: Array<String>) {
  // Read number of testcases from standard input.
  var T = readLine()!!.toInt()
  for (tc in 1..T) {
    // Read integers n and m from the standard input.
    var (N, M) = readLine()!!.split(" ").map { it.toInt() }
    // Read n integers from the standard input and save them in the array `C`.
    var C = readLine()!!.split(" ").map { it.toInt() }

    // Declare a variable for sum and set it to 0.
    var sum = 0
    // Loop through the array `bag` and sum its values.
    for (i in 0..N - 1) {
      sum += C[i]
    }
    // Compute the value of sum modulo m.
    var modulo = sum % M
    // Print result to the standard output.
    println("Case #$tc: $modulo")
  }
}
